### PR TITLE
Add ability to find "similar" avatars in the media browser

### PR DESCRIPTION
**Pull request by netpro2k**
Adds "find similar" buttons to avatar listing results in the media browsers. Currently this is implemented in the backend to just find avatars that share the same parent avatar, but this can be updated later to actually use a more interesting metric of similarity.

Requires backend changes here: https://github.com/mozilla/reticulum/pull/249

### CODE REVIEW

File: src/react-components/media-browser.js [hunk 1]
```diff
@@ -129,6 +129,7 @@ class MediaBrowser extends Component {
     const urlSource = this.getUrlSource(searchParams);
     newState.showNav = !!(searchParams.get("media_nav") !== "false");
     newState.selectAction = searchParams.get("selectAction") || "spawn";
+    newState.query = searchParams.get("q") || "";
```
Comment (gfodor) [inline src/react-components/media-browser.js]: it looks like this should be reflected on line 128. do we want to discard the value in `state` here in all cases? i haven't looked at this code in a while

File: src/react-components/media-browser.js [hunk 2]
```diff
@@ -265,7 +272,8 @@ class MediaBrowser extends Component {
       }
     };
 
-    const activeFilter = searchParams.get("filter") || (!searchParams.get("q") && "");
+    const activeFilter =
+      searchParams.get("filter") || (searchParams.get("similar_to") && "similar") || (!searchParams.get("q") && "");
```
Comment (gfodor) [inline src/react-components/media-browser.js]: whoa I found this line confusing. i guess a bunch of `if`s is worse tho.

File: src/react-components/media-browser.js [hunk 3]
```diff
@@ -401,8 +409,13 @@ class MediaBrowser extends Component {
                     {s.text}
                   </a>
                 ))}
-              </div>
-            )}
+              {activeFilter === "similar" && (
+                <a className={classNames(styles.facet, "selected")}>
+                  Similar to: &quote;{searchParams.get("similar_name")}&quote;
```
Comment (gfodor) [inline src/react-components/media-browser.js]: should be moved into strings file

File: src/react-components/media-tiles.js [hunk 4]
```diff
@@ -172,28 +174,41 @@ class MediaTiles extends Component {
...
+              title="Edit"
             >
-              <FontAwesomeIcon icon={faClone} />
+              <FontAwesomeIcon icon={faPencilAlt} />
             </StateLink>
           )}
+          {entry.type === "avatar_listing" && (
+            <StateLink
+              onClick={e => {
+                e.preventDefault();
```
Comment (gfodor) [inline src/react-components/media-tiles.js]: i don't think you need to do this w `StateLink`

File: src/react-components/media-browser.js (hunk 1 above)
Comment (netpro2k) [reply src/react-components/media-browser.js]: From what I can tell `getStoreAndHistoryState` is called only on page load and whenever the store has been updated, in both cases it seems like we want to overwrite the state. I will combine this with the line above though.

Comment (netpro2k) [reply src/react-components/media-browser.js]: From what I can tell `getStoreAndHistoryState` is called only on page load and whenever the store has been updated, in both cases it seems like we want to overwrite the state. I will combine this with the line above though.

**Pull request merged**